assets/js/script.js

```diff
@@ -37,4 +100,44 @@ const getAnimals = (data) => {
     })
 };
 
-$tokenButton.on('click', getToken);
\ No newline at end of file
+$tokenButton.on('click', getToken);
+
+//Loads the favorites on page-load to populate #favorites from previously saved favorites.
+loadFavorites();
+
+//When the heart icon is clicked, it adds the name and ID to the favorites.
+$(".icon").on("click", function() {
+    var $petContent = $(this).closest(".card-content");
+    
+    addFavorites(
+        $petContent.children(".name").text(),
+        $petContent.attr("data-id")
+    );
+});
+
+//When a favorites button is pressed it uses the attached ID to query the specifc animal.
+$("#favorites").on("click", "button", function() {
+    getToken()
+
+    let TOKEN = localStorage.getItem('token');
+
+    $.ajax({
+        type: "GET",
+        url: `https://api.petfinder.com/v2/animals/${$(this).attr("data-id")}`,
+        headers: {
+            Authorization: `Bearer ${TOKEN}`
+        },
+        success: function(data) {
+            /*****************************TO DO*********************************
+            *On success the data needs to be used to open and populate a modal.*
+            *******************************************************************/
+            console.log(data)
+        },
+        dataType: "json"
+    })
+});
+            getAnimals(data);
+        },
+        dataType: "json"
+      })
+};
```

Comment:
at bottom of the script add in `getToken()` and `setInterval(() => { getToken() }, (3600 * 1000));`